docs/Procedure/Documentation/Step_Registry_Documentation_Guide.md

```diff
@@ -0,0 +1,155 @@
+- If this step utilizes any specific (important) files in the test repository, provide a link to the file in GitHub
+
+The documentation doesn't need to exhaustive, but, there should be a general description of how the step works and any outside resources it may utilize. Aim to give someone enough information so that they don't have to step through the code to figure out what your new step does and why.
+
+#### Requirements
+This section can provide very useful information to anybody who may need to utilize or debug this step. Make sure to explain why the following requirements are needed and how to get them.
+
+##### Variables<!-- omit from toc -->
+For each environment variable in the `env` list of your new step should be outlined. Use the following as a guide for how to document these variables. We will start with an example configuration file:
+
+<sub><sup>`interop-tooling-new-tool-ref.yaml`</sup></sub>
+```yaml
+ref:
+  as: interop-tooling-new-tool
+  from: cli
+  commands: interop-tooling-new-tool-commands.sh
+  env:
+  - name: USERNAME
+    documentation: Some username for some reason
+  - name: PASSWORD
+    documentation: Some password for some reason
+    default: "reallyC00lP@ssw0rd"
+  documentation: |-
+    Do something cool in this new tool...
+```
+This configuration file shows that we have defined two required environment variable in the `env` stanza. These two environment variables have a brief description that is used in the [automatically generated documentation](#automatically-generated-documentation). While it may seem simple to just check the config for these requirements, it is nice to have them in the `README.md` for ease of access when somebody is reading the documentation.
+
+To document these two variables, we will uses the following format:
+
+```markdown
+### Variables
+
+- `USERNAME` 
+  - **Definition**: The username used in some part of the script. VERY IMPORTANT
+  - **If left empty**: The script will fail.
+- `PASSWORD`
+  - **Definition**: The password associated with the USERNAME variable.
+  - **If left empty**: A default password of "reallyC00lP@ssw0rd" will be used.
+```
+Feel free to add additional information, the information above is mostly a suggestion.
+
+##### Infrastructure<!-- omit from toc -->
+There isn't a specific format associated with this section, but it is important. Please make a list of any infrastructure (inside or out of the test cluster) required to run this step. Some examples of this could be:
+- A provisioned test cluster
+- A container running in the test cluster
+- Specific operator(s) that need to installed
+- A resource outside of openshift (specific repository or access to a server)
+
+##### Credentials<!-- omit from toc -->
+For each item in the `credentials` stanza of the step's configuration should be outlined. Use the following as a guide for how to document these credentials. We will start with an example configuration file:
```

Comment:
change `Use the following as a guide for how to document` to `Use the following as a guide for documenting`